tests/mcs_benchmark_data_test/assertions.py

```diff
@@ -33,3 +35,97 @@ def assert_submission_models(*, benchmark_id: str, submission_id: str, models):
     samples = [model for model in models if isinstance(model, SubmissionSample)]
     assert len(samples) > 3
     assert all(sample.submission_uri == submission.uri for sample in samples)
+
+
+def assert_benchmark_pipeline_compiles(*, root_path: Path, benchmark_name: str):
+
+    paths = [
+        Path(root_path / "data" / benchmark_name / "datasets" / dataset_type.value)
+        for dataset_type in DatasetType
+    ]
+
+    paths += [
+        Path(root_path / "test_data" / benchmark_name / "datasets" / dataset_type.value)
+        for dataset_type in DatasetType
+    ]
+    path_to_pipeline = Path(
+        root_path / "mcs_benchmark_data" / "pipelines" / benchmark_name
+    )
+    path_to_tests = Path(
+        root_path / "tests" / "mcs_benchmark_data_test" / "pipelines" / benchmark_name
+    )
+    path_to_metadata = Path(root_path / "data" / Path(benchmark_name) / "metadata.json")
+    paths.append(path_to_pipeline)
+    paths.append(path_to_tests)
+    for path in paths:
+        assert path.exists()
+        if path == path_to_pipeline:

```

Comment:
Rather than have these if branches, you can do something like have `paths` be tuples of (`Path`, callable_check).


For `.py` files, `callable_check` will be a function that takes the path (the first element of the tuple) and compiles the file.
For `.jsonl`, one that parses the .jsonl?

You can reuse functions for callable check e.g.,
```
def check_py(path: Path):
   py_compile...
```
Then
```
paths.append((path_to_py_file, check_py))
```
Then the loop over `paths` does:
```
for path, callable_check in paths:
    assert path.exists()
    callable_check(path)
```